src/runtime/auto-prompt-manager.js

```diff
@@ -410,6 +419,33 @@ export class AutoPromptManager {
     return true;
   }
 
+  /**
+   * Determines what Audience Action prompt type should be shown.
+   *
+   * Show the first AutoPromptType passed in from Audience Actions.
+   * @param {{
+   *   article: (!./entitlements-manager.Article)
+   * }} article
+   * @return {!AutoPromptType|undefined}
+   */
+  getAutoPromptType_(article) {
+    const audienceActions = article.audienceActions?.actions || [];
+
+    const potentialActions = audienceActions.filter(
```

Comment:
Since we just need one match, we could use the `find()` method, which could simplify the code a bit
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Reply:
Done.